app/controllers/login.js

```diff
@@ -1,30 +1,19 @@
 import Controller from '@ember/controller';
 import { inject as service } from '@ember/service';
-import { set } from '@ember/object';
 
 export default Controller.extend({
   session: service(),
-
-  errors: null,
+  router: service(),
 
   email: '',
   password: '',
 
-  init() {
-    this._super(...arguments);
-    set(this, 'errors', []);
-  },
-
   actions: {
-    login(email, password) {
-      return this.session
-        .authenticate('authenticator:conduit', { email, password })
-        .then(() => {
-          this.transitionToRoute('home');
-        })
-        .catch(normalizedErrors => {
-          set(this, 'errors', normalizedErrors);
-        });
+    async logIn() {
```

Comment:
async actions can have a whole load of gotyas (especially in testing) so I would strongly advise against using them 🙈 it is also harder to disable the login button while you are logging in (preventing the double click issue)

If you make this an ember concurrency task you will be able to save issues with testing and easily disable the button while the task is in progress

Reply:
I thought all of that runloop/testing stuff was resolved when Ember shipped the Microtasks integration in 3.4? I'm going off of the very long discussion in [this Discuss post](https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736/36)

I do get the point about disabling the button though. I'm trying to remember our conversation about dependencies. We've decided it's okay to use ember-concurrency?

Reply:
yes we said that if we're doing any async actions then we should be using ember-concurrency 👍

Reply:
I agree with @mansona here :)
Let's use `ember-concurrency` where appropriate?

Reply:
okely dokely 👍 will update. Thank!

Reply:
@Alonski @mansona this should be good for another 👀 now